src/components/20-molecules/datepicker/index.js

```diff
@@ -104,15 +103,17 @@ export const parseAndFormatAllowedYears = (allowedyears = [], setYear) => {
 };
 
 // gather all native Date-object setters in one place
-const overrideDate = (year, month, day, date) => {
+const overrideDate = (year, month, day, date, status = {}) => {
   let _day = day;
 
   if (typeof year === 'number' && year >= 0) {
     date.setFullYear(year);
+    status.set = true;
```

Comment:
Do you need a object `status` here? is a boolean variable sufficient?

Reply:
This function returns a `Date` instance. If we don't want to change how the function is _used_, we have no choice but to pass in a modifiable object, and set a key in that to propagate that info outwards.

This way of doing things is the least invasive option, IMHO, that's why I chose it.